src/wallet/coinselection.cpp

```diff
@@ -156,7 +156,7 @@ std::optional<SelectionResult> SelectCoinsBnB(std::vector<OutputGroup>& utxo_poo
     for (const size_t& i : best_selection) {
         result.AddInput(utxo_pool.at(i));
     }
-    result.ComputeAndSetWaste(CAmount{0});
+    result.ComputeAndSetWaste(cost_of_change, cost_of_change, CAmount{0});
```

Comment:
In "wallet: use GetChange() when computing waste" (45ee9d3a737cc23babb9200c4d75d7dae4dc6700):

Shouldn't the first value here be the maximum of `cost_of_change`  and  `dust_threshold`? I think that the former can be lower than the latter for feerates below discard_feerate.

Reply:
Technically yes, but it doesn't matter in this case because BnB would never have change more than `cost_of_change`.

I'd say a better way to fix it is to use `min_viable_change` for the upper bound of BnB. But I don't want to do expand the scope of this PR. Can we leave it for the follow up?

Reply:
Ah right, and since this call originates from `SelectCoinsBnB(…)` that's sufficient of course.